firestore/app/src/main/java/com/google/example/firestore/DocSnippets.java

```diff
@@ -128,18 +130,28 @@ public void setup() {
         // [END get_firestore_instance]
 
         // [START set_firestore_settings]
-        FirebaseFirestoreSettings settings = new FirebaseFirestoreSettings.Builder()
-                .setPersistenceEnabled(true)
-                .build();
+        FirebaseFirestoreSettings settings = 
+        new FirebaseFirestoreSettings.Builder(db.getFirestoreSettings())
+            // Use memory-only cache
+            .setLocalCacheSettings(MemoryCacheSettings.newBuilder().build())
+            // Use persistent disk cache (default)
+            .setLocalCacheSettings(PersistentCacheSettings.newBuilder()
+                                    .build())
+            .build();
         db.setFirestoreSettings(settings);
         // [END set_firestore_settings]
     }
 
     public void setupCacheSize() {
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
         // [START fs_setup_cache]
-        FirebaseFirestoreSettings settings = new FirebaseFirestoreSettings.Builder()
-                .setCacheSizeBytes(FirebaseFirestoreSettings.CACHE_SIZE_UNLIMITED)
-                .build();
+        FirebaseFirestoreSettings settings = 
+        new FirebaseFirestoreSettings.Builder(db.getFirestoreSettings())
+            .setLocalCacheSettings(PersistentCacheSettings.newBuilder()
+                                    // Set size to 1 MB
+                                    .setSizeBytes(1_000_000)
```

Comment:
Sorry for the late comment.

This needs to be fixed too: 1_000_000 is not 1MB. And we should use 100 MB here.

Reply:
Fixed in #456.